File: src/lib/limits.ts
/** Limites partilhados das métricas corporais.
 *
 * Antes estes intervalos estavam duplicados (Onboarding, Perfil, PesoDetail e o
 * backend) com valores diferentes — um perfil válido num sítio era rejeitado
 * noutro. Ficam aqui, alinhados com backend/app/data/schemas.py. */

export interface Range {
  min: number
  max: number
}

export const WEIGHT_KG: Range = { min: 25, max: 400 }
export const HEIGHT_CM: Range = { min: 80, max: 250 }
export const AGE: Range = { min: 10, max: 120 }
/** % de gordura corporal plausível (Katch-McArdle). */
export const BODY_FAT_PCT: Range = { min: 3, max: 70 }

export function inRange(value: number, range: Range): boolean {
  return Number.isFinite(value) && value >= range.min && value <= range.max
}

File: ios/Macros/Macros/Lib/Limits.swift
//
//  Limits.swift
//  Macros
//
//  Limites partilhados das métricas corporais — equivalente de src/lib/limits.ts.
//

import Foundation

struct ValueRange {
    let min: Double
    let max: Double
    func contains(_ value: Double) -> Bool {
        value.isFinite && value >= min && value <= max
    }
}

enum Limits {
    static let weightKg = ValueRange(min: 25, max: 400)
    static let heightCm = ValueRange(min: 80, max: 250)
    static let age = ValueRange(min: 10, max: 120)
    /// % de gordura corporal plausível (Katch-McArdle).
    static let bodyFatPct = ValueRange(min: 3, max: 70)

    // Limites usados no onboarding (ligeiramente mais apertados que os limites
    // absolutos acima, tal como no formulário original em Onboarding.tsx).
    static let onboardingAge = ValueRange(min: 14, max: 100)
    static let onboardingHeight = ValueRange(min: 120, max: 230)
    static let onboardingWeight = ValueRange(min: 35, max: 250)
}
